packages/teams-js/src/private/externalAppCardActionsForCEA.ts

```diff
@@ -0,0 +1,99 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { validateId } from '../internal/utils';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../public/constants';
+import { runtime } from '../public/runtime';
+import { ActionOpenUrlError, ActionOpenUrlType, ActionSubmitError, IAdaptiveCardActionSubmit } from './interfaces';
+
+/**
+ * Updated to constants file: v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const externalAppCardActionsTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+export namespace externalAppCardActionsForCEA {
+  /**
+   * @beta
+   * @hidden
+   * Delegates an Adaptive Card Action.OpenUrl request to the host for the application with the provided app ID.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application the request is intended for. This must be a UUID
+   * @param conversationId To tell the bot what conversation the calls are coming from
+   * @param url The URL to open
+   * @returns Promise that resolves to ActionOpenUrlType indicating the type of URL that was opened on success and rejects with ActionOpenUrlError if the request fails
+   */
+  export function processActionOpenUrl(appId: string, conversationId: string, url: URL): Promise<ActionOpenUrlType> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    validateId(appId, new Error('App id is not valid.'));
+    validateId(conversationId, new Error('conversation id is not valid.'));
+    return sendMessageToParentAsync<[ActionOpenUrlError, ActionOpenUrlType]>(
+      getApiVersionTag(
+        externalAppCardActionsTelemetryVersionNumber,
+        ApiName.ExternalAppCardActionsForCEA_ProcessActionOpenUrl,
+      ),
+      'externalAppCardActionsForCEA.processActionOpenUrl',
```

Comment:
>'externalAppCardActionsForCEA.processActionOpenUrl' [](http://example.com/codeflow?start=6&length=51)

Can this reference `ApiName.ExternalAppCardActionsForCEA_ProcessActionOpenUrl` rather than duplicating the string?